### DISCUSSION TITLE
Should I create multiple threads to support multiple windows?

### DISCUSSION THREAD

**Discussion by xland** [mikke89/RmlUi#405] [Q&A] [answered]
Because there is an infinite loop in the entry function
```c++
bool running = true;
while (running)
{
    running = Backend::ProcessEvents(context, &Shell::ProcessKeyDownShortcuts);
    context->Update();
    Backend::BeginFrame();
    context->Render();
    Backend::PresentFrame();
}
```
So should I create multiple threads to support multiple windows?
Thanks very much.

**Comment by mikke89** [comment, marked as answer]
The backends should be considered basic examples for the most common use cases. If you need extra features, like multi-window in your case, then I recommend that you step in and modify the backend (and the loop) to your own needs.

It would probably be a lot easier if you stick to a single thread, so I recommend starting there unless you have some experience with multi-threaded environments. Remember that any access into RmlUi needs to be synchronized when using multiple threads.

**Reply by xland** [reply]
Thanks very much.I'm trying to make my own backend.
